app/validators/user_validator.rb

```diff
@@ -0,0 +1,19 @@
+class UserValidator < ActiveModel::Validator
+  def validate(record)
+    if record.username.blank?
+      record.errors.add(:username, "The username is required.")
+    elsif record.username.length > 50
+      record.errors.add(:username, "You cannot input more 50 characters.")
+    end
+    if record.password.blank?
```

Comment:
The `UserValidator` class is using manual validation which can be error-prone and hard to maintain. Consider using ActiveModel validations for better maintainability.